src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -244,7 +246,11 @@ private void OutputBaseSize(ref ObjectDataBuilder objData)
             int minimumObjectSize = pointerSize * 3;
             int objectSize;
 
-            if (_type.IsDefType)
+            if (_type is CanonBaseType)
+            {
+                objectSize = pointerSize;
```

Comment:
The only difference is that we'll emit a valuetype padding for __Universal canon. Made it a TODO for now